### PR TITLE
Added emits parameters to rocket modules

### PR DESCRIPTION
**Pull request by brrmorre**
**Related issue**: 

**Type of change**: other enhancement

**Impact**: no functional change

**Development Phase**: implementation

**Release Notes**

Added emits parameters to rocket modules.

### CODE REVIEW

File: src/main/scala/rocket/HellaCache.scala [hunk 1]
```diff
@@ -179,15 +179,29 @@ abstract class HellaCache(staticIdForMetadataUseOnly: Int)(implicit p: Parameter
...
     name          = s"Core ${staticIdForMetadataUseOnly} DCache",
     sourceId      = IdRange(0, 1 max cfg.nMSHRs),
-    supportsProbe = TransferSizes(cfg.blockBytes, cfg.blockBytes))))
-
-  protected def mmioClientParameters = Seq(TLMasterParameters.v1(
+    emits = TLMasterToSlaveTransferSizes(
+      arithmetic = TransferSizes(4, p(XLen) / 8),
+      logical = TransferSizes(4, p(XLen) / 8),
+      get = TransferSizes(1, p(XLen) / 8),
+      putFull = TransferSizes(1, p(XLen) / 8)
```
Comment (hcook) [inline src/main/scala/rocket/HellaCache.scala:189]: Cacheable only emits Acquires

Comment (hcook) [reply src/main/scala/rocket/HellaCache.scala:189]: of size cfg.blockBytes

File: src/main/scala/rocket/ICache.scala [hunk 2]
```diff
@@ -57,10 +57,13 @@ class ICache(val icacheParams: ICacheParams, val staticIdForMetadataUseOnly: Int
...
   val useVM = p(TileKey).core.useVM
-  val masterNode = TLClientNode(Seq(TLMasterPortParameters.v1(
-    clients = Seq(TLMasterParameters.v1(
+  val masterNode = TLClientNode(Seq(TLMasterPortParameters.v2(
+    masters = Seq(TLMasterParameters.v2(
       sourceId = IdRange(0, 1 + icacheParams.prefetch.toInt), // 0=refill, 1=hint
-      name = s"Core ${staticIdForMetadataUseOnly} ICache")),
+      name     = s"Core ${staticIdForMetadataUseOnly} ICache",
+      emits    = TLMasterToSlaveTransferSizes(
+        get = TransferSizes(icacheParams.fetchBytes, icacheParams.fetchBytes)
```
Comment (brrmorre) [inline src/main/scala/rocket/ICache.scala:65]: Henry said this seems correct

File: src/main/scala/tilelink/RationalCrossing.scala [hunk 3]
```diff
@@ -20,7 +20,7 @@ class TLRationalCrossingSource(implicit p: Parameters) extends LazyModule
 
   lazy val module = new LazyModuleImp(this) {
     (node.in zip node.out) foreach { case ((in, edgeIn), (out, edgeOut)) =>
-      val bce = edgeIn.manager.anySupportAcquireB && edgeIn.client.anySupportProbe
+      val bce = edgeIn.diplomaticClaimsMasterToSlave.acquireB && edgeIn.diplomaticClaimsSlaveToMaster.probe
```
Comment (brrmorre) [inline src/main/scala/tilelink/RationalCrossing.scala:23]: Don't do this. Its wrong

Comment (brrmorre) [reply src/main/scala/tilelink/RationalCrossing.scala:23]: This type of change is unnecessary

Comment (brrmorre) [reply src/main/scala/tilelink/RationalCrossing.scala:23]: I will leave the code as is

File: src/main/scala/tilelink/Parameters.scala [hunk 4]
```diff
@@ -1128,13 +1128,15 @@ class TLMasterPortParameters private(
     }
   }
 
-  // Diplomatically determined operation sizes emitted by all inward Masters
+  // Diplomatically determined operation sizes emitted/supported by all inward Masters
   // as opposed to expectsVipChecker which generate circuitry to check which specific addresses
-  val allEmitClaims = masters.map(_.emits).reduce( _ intersect _)
+  val allEmitClaims    = masters.map(_.emits)   .reduce( _ intersect _)
```
Comment (jchang0) [inline src/main/scala/tilelink/Parameters.scala:1133]: I'm not sure I see a usage case for this. Should probably remove it (and the intersect for support also).

File: src/main/scala/rocket/ICache.scala [hunk 5]
```diff
@@ -58,10 +58,14 @@ class ICache(val icacheParams: ICacheParams, val staticIdForMetadataUseOnly: Int
...
-  val masterNode = TLClientNode(Seq(TLMasterPortParameters.v1(
-    clients = Seq(TLMasterParameters.v1(
+  val masterNode = TLClientNode(Seq(TLMasterPortParameters.v2(
+    masters = Seq(TLMasterParameters.v2(
       sourceId = IdRange(0, 1 + icacheParams.prefetch.toInt), // 0=refill, 1=hint
-      name = s"Core ${staticIdForMetadataUseOnly} ICache")),
+      name     = s"Core ${staticIdForMetadataUseOnly} ICache",
+      emits    = TLMasterToSlaveTransferSizes(
+        get = TransferSizes(icacheParams.blockBytes, icacheParams.blockBytes),
+        hint = TransferSizes(icacheParams.blockBytes, icacheParams.blockBytes)
```
Comment (brrmorre) [inline src/main/scala/rocket/ICache.scala:67]: hint is turned off when prefetch is turned off, should add that

**Comment by brrmorre** [comment]
copied over to https://github.com/chipsalliance/rocket-chip/pull/2683

**Pull request closed without merging**